coalib/bearlib/languages/documentation/DocumentationComment.py

```diff
@@ -103,6 +123,35 @@ def _parse_documentation_with_symbols(self,
             section is a named tuple of either ``Description``, ``Parameter``,
             ``ExceptionValue`` or ``ReturnValue``.
         """
+
+        def startswith(line, starter):
+            if isinstance(starter, type(None)) or line.startswith(starter):
+                if isinstance(starter, tuple):
+                    for s in starter:
+                        if line.startswith(s):
+                            return s
+                else:
+                    return starter
+            return False
+
+        def _extract_typeRef(ref_identifiers, typeRef):
+            """
+            ":mod:`" --> module
+            ":func:`" --> function
+            ":data:`" --> data
+            ":constant:`" --> constant
+            ":class:`" --> class
+            ":meth:`" --> method
+            ":attr:`" --> attribute
+            ":exc:`" --> exception
+            ":obj:`" --> object
+            """
+            typeRef = typeRef.strip()
```

Comment:
I have a gut feeling that this function can be optimised, do we need to? and how?